docs/latex/Inhalt/03_Inhalt/04_unit_tests.tex

```diff
@@ -1,55 +1,60 @@
+\paragraph{Professional} Um dieses Kriterium zu erfüllen, wurde darauf geachtet, dass der Testcode möglichst leicht verständlich ist. Hierzu wurden beispielsweise die Namen der einzelnen Tests nach einem bestimmten immer gleichen Muster gewählt: Jede Testmethode beginnt mit dem Wort \enquote{should} und beschreibt das erwartete Verhalten des zu testenden Codes, zum Beispiel \href{https://github.com/anditru/quickie/blob/bb41442c7f1ffbfcd3117cd86a40f7932e543a33/2-quickie-application/src/test/java/org/pinkcrazyunicorn/quickie/application/recipe/MatchingServiceTest.java#L21}{\code{shouldFindExpectedMatchingRecipes}}. Außerdem wurde das jeweils zu testende Objekt stets mit \code{codeUnderTest} bezeichnet. Ferner folgen alle Tests wie zuvor erwähnt der AAA-Normalform, bestehen also stets aus den gleichen drei Phasen \enquote{Arrange}, \enquote{Act} und \enquote{Assert} besteht. Für die letzte Phase wurde außerdem die Library AssertJ verwendet, welche eine fluent \ac{API} für besser lesbare Assertions bietet.
 
 \section{Code-Coverage}
 \label{sec:code-coverage}
-Wie eingangs erwähnt beschränken sich die implementierten Unit Tests im Wesentlichen auf die Applikationsschicht, da dort der Großteil der Businesslogik vorhanden ist. Hier wird jedoch nach beiden gängigen Standards eine sehr hohe Code-Coverage erreicht. \autoref{tab:code-coverage} zeigt eine Messung der Code-Coverage der \acs{IDE} Intellij IDEA für das Package \code{org.pinkcrazyunicorn.quickie.application}, welches den gesamten Code der Applikationsschicht enthält. So beträgt die Line-Coverage hier 98\% und die aussagekräftigere Branch-Coverage 80\%.
+Wie eingangs erwähnt beschränken sich die implementierten Unit Tests im Wesentlichen auf die Applikationsschicht, da dort der Großteil der Businesslogik vorhanden ist. Hier wird jedoch nach beiden gängigen Standards eine sehr hohe Code-Coverage erreicht. \autoref{tab:code-coverage} zeigt eine Messung der Code-Coverage der \acs{IDE} Intellij IDEA für das Package \code{org.pinkcrazyunicorn.quickie.application}, welches den gesamten Code der Applikationsschicht enthält. So beträgt die Line-Coverage hier 98\% und die aussagekräftigere Branch-Coverage 80\%. 
+
+In \autoref{tab:code-coverage-full} kann die Code-Coverage über das gesamte Projekt gesehen werden. Einzelne weniger relevante Subpackages werden hier nicht dargestellt. In beiden Tabellen wird mit \enquote{NaN} gezeigt, dass keine Zeilen oder Branches vorhanden sind, die gemessen werden könnten, wie es zum Beispiel bei Interfaces der Fall ist. Über das gesamte Projekt besteht eine Line-Coverage von 17\% und eine Branch-Coverage von 57\%. Besonders gut ist dabei das Einhalten der Dependency Rule zu sehen: Da nur die Applikationsschicht sowie eine Klasse der Adapterschicht getestet wurden, hat die Pluginschicht keinerlei Testabdeckung. Der Domaincode hingegen ist zu teilen getestet, da eine Abhängigkeit auf diesen durch die Applikationsschicht besteht.
```

Comment:
"In \autoref{tab:code-coverage-full} kann die Code-Coverage über das gesamte Projekt gesehen werden" Da sind wir wieder beim Thema Passiv. -> "\autoref{tab:code-coverage-full} zeigt die Code-Coverage über das gesamte Projekt.

Reply:
Hatte eigentlich den Grund, dass ich eine andere Formulierung wollte als im Absatz drüber

Reply:
Dachte ich mir fast, aber lieber zwei mal die gleiche Formulierung als das.